tools/run_tests/python_utils/port_server.py

```diff
@@ -182,4 +186,16 @@ def do_GET(self):
 class ThreadedHTTPServer(ThreadingMixIn, HTTPServer):
   """Handle requests in a separate thread"""
 
+def _flush_output_forever():
+  while True:
+    time.sleep(1)
+    sys.stderr.flush()
+    sys.stdout.flush()
+
+output_flushing_thread = threading.Thread(target=_flush_output_forever)
+output_flushing_thread.daemon = True
+output_flushing_thread.start()
+
+dns_server.start_local_dns_server_in_background(_DNS_SERVER_PORT)
```

Comment:
@nicolasnoble recommended this solution, since we need a hermetic DNS server in the tests.  I suppose we could have the tests run yet another binary, but that seems unnecessarily complicated.